internal/servers/worker/session.go

```diff
@@ -331,6 +349,28 @@ func (w *Worker) makeSessionCloseInfo(
 	return result
 }
 
+// makeFakeSessionCloseInfo makes a "fake" makeFakeSessionCloseInfo, intended
+// for use when we can't contact the controller.
+func (w *Worker) makeFakeSessionCloseInfo(
+	closeInfo map[string]string,
+) map[string][]*pbs.CloseConnectionResponseData {
+	if closeInfo == nil {
+		// Should never happen, panic if it does. Results will be
+		// undefined.
+		panic(errMakeSessionCloseInfoNilCloseInfo)
+	}
+
+	result := make(map[string][]*pbs.CloseConnectionResponseData)
+	for connectionId, sessionId := range closeInfo {
+		result[sessionId] = append(result[sessionId], &pbs.CloseConnectionResponseData{
+			ConnectionId: connectionId,
+			Status:       pbs.CONNECTIONSTATUS_CONNECTIONSTATUS_CLOSED,
+		})
+	}
+
+	return result
+}
+
```

Comment:
In setCloseTimeForResponse below, do we really want it to always mark the connection's close time as now? What if it was already closed -- we probably don't want to keep moving the close time forward, but want to preserve it as it was.

Reply:
Sure, just want to point out though that this is mainly just carrying forward existing functionality. This logic does not exist currently, IIRC. Here's the [existing code from v0.3.0](https://github.com/hashicorp/boundary/blob/407a21991aa7dc550967720466d4b10c2e02ee1c/internal/servers/worker/session.go#L288-L294).

Reply:
Discussed offline that this is something that we want to defer to post-release, given that it would ultimately be new behavior. Additionally, connection close requests at the controller level are something that could probably use a bit more work in general.